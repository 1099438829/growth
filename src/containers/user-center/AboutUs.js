import React from 'react';
import { Text } from 'react-native';

const AboutUs = () => (
  <Text>AboutUs</Text>
);

export default AboutUs;
