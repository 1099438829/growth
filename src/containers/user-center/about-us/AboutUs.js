import React from 'react';
import { Linking, ScrollView, Text, View } from 'react-native';
import { List, ListItem } from 'react-native-elements';
import AppStyle from '../../../theme/styles';

const contributors = [{
  avatar: 'https://facebook.github.io/react/img/logo_og.png',
  name: 'Name Placeholder',
  description: 'Placeholder',
}, {
  avatar: 'https://facebook.github.io/react/img/logo_og.png',
  name: 'Name Placeholder 1',
  description: 'Placeholder 1',
}];

export const Label = props => (
  <View style={{ paddingLeft: 20, marginTop: 12, marginBottom: -6 }}>
    <Text style={{ fontSize: 14 }}>
      {props.text}
    </Text>
  </View>
);

Label.propTypes = {
  text: React.PropTypes.string.isRequired,
};

const AboutUs = () => (
  <ScrollView style={AppStyle.userCenterBasisStyle}>
    <Label text="Growth" />
    <List>
      <ListItem
        title={'开源 - GitHub'}
        rightIcon={{ name: 'github', type: 'font-awesome' }}
        onPress={() => Linking.openURL('https://github.com/phodal/growth-ng')}
      />
      <ListItem
        title={'QQ 交流群: 529600495'}
        hideChevron
      />
      <ListItem
        title={'论坛: https://forum.growth.ren'}
        onPress={() => Linking.openURL('https://forum.growth.ren')}
        hideChevron
      />
    </List>
    <Label text="贡献者" />
    <List>
      {contributors.map(contributor =>
        <ListItem
          key={contributor.name}
          title={contributor.name}
          subtitle={contributor.description}
          avatar={{ uri: contributor.avatar }}
          roundAvatar
          hideChevron
        />,
      )}
    </List>
  </ScrollView>
);

export default AboutUs;
