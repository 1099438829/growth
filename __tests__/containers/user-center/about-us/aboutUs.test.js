// import 'react-native';
import React from 'react';
import { Linking } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as chai from 'chai';
import ListItem from 'react-native-elements/src/list/ListItem';

import AboutUs, { Label } from '../../../../src/containers/user-center/about-us/AboutUs';

describe('test user center / about us', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <AboutUs />,
    );

    const treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
  });

  it('test for pressing links', () => {
    const spy = jest.spyOn(Linking, 'openURL');

    const wrapper = shallow(<AboutUs />);

    wrapper.find({ title: '开源 - GitHub' }).props().onPress();
    wrapper.find({ title: '论坛: https://forum.growth.ren' }).props().onPress();

    expect(spy).toHaveBeenCalledTimes(2);
  });

  it('test for children components', () => {
    const wrapper = shallow(<AboutUs />);

    chai.expect(wrapper.find(ListItem)).to.have.length(5);
    chai.expect(wrapper.find({ title: '开源 - GitHub' })).to.have.length(1);
    chai.expect(wrapper.find({ title: 'QQ 交流群: 529600495' })).to.have.length(1);
    chai.expect(wrapper.find({ title: '论坛: https://forum.growth.ren' })).to.have.length(1);
  });

  it('test for rendering Label', () => {
    const tree = renderer.create(
      <Label text="LabelText" />,
    );

    const treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
  });
});
