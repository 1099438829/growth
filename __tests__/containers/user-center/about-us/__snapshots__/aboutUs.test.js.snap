// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test user center / about us renders correctly 1`] = `null`;

exports[`test user center / about us test for rendering Label 1`] = `
<View
  style={
    Object {
      "marginBottom": -6,
      "marginTop": 12,
      "paddingLeft": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    LabelText
  </Text>
</View>
`;
